serpy/serializer.py

```diff
@@ -100,29 +143,71 @@ def _serialize(self, obj, fields):
                 if required or result is not None:
                     if call:
                         result = result()
-                    if to_value:
-                        result = to_value(result)
+                    if to_repr:
+                        result = to_repr(result)
             v[name] = result
 
         return v
 
-    def to_value(self, obj):
-        fields = self._compiled_fields
+    def _deserialize(self, data, fields):
+        v = self._cls()
+        for name, setter, to_internal, call, required, pass_self in fields:
+            if pass_self:
+                setter(self, v, data[name])
+            else:
+                if required:
+                    value = data[name]
+                else:
+                    value = data.get(name)
+                if to_internal and (required or value is not None):
+                    value = to_internal(value)
+                setter(v, value)
+        return v
+
+    def to_representation(self, obj):
+        fields = self._compiled_read_fields
         if self.many:
             serialize = self._serialize
             return [serialize(o, fields) for o in obj]
         return self._serialize(obj, fields)
 
+    def to_internal_value(self, data):
+        fields = self._compiled_write_fields
+        if self.many:
+            deserialize = self._deserialize
+            return [deserialize(o, fields) for o in data]
+        return self._deserialize(data, fields)
+
     @property
-    def data(self):
+    def representation(self):
         """Get the serialized data from the :class:`Serializer`.
 
-        The data will be cached for future accesses.
+        The representation will be cached for future accesses.
+        """
+        # Cache the representation for next time .representation is called.
+        if self._representation is None:
+            self._representation = self.to_representation(self._initial_obj)
+        return self._representation
+
+    @property
+    def data(self):
+        warnings.warn(
+            "`.data` property is deprecated, use `.representation` instead",
+            DeprecationWarning,
+            stacklevel=2
+        )
+        return self.representation
+
+    @property
+    def internal_value(self):
```

Comment:
how about calling this `deserialized_value` instead. Open to suggestions for other names though